doc/changes/latest.inc

```diff
@@ -68,4 +68,4 @@ Bugs
 
 API changes
 ~~~~~~~~~~~
-- Nothing yet
+- :meth:`mne.Info.pick_channels` has been deprecated. Use ``inst.pick_channels`` to pick channels from Raw|Epochs|Evoked. Use :func:`mne.pick_info` to pick channels from :class:`mne.Info` (:gh:`10039` by `Mathieu Scheltienne`_)
```

Comment:
This saves us some work later for when we actually remove it (the link will be dead at that point). Plus for most deprecated functions we remove them from the API page, so for consistency it's not bad to avoid linking to the docs.
```suggestion
- ``mne.Info.pick_channels`` has been deprecated. Use ``inst.pick_channels`` to pick channels from Raw|Epochs|Evoked. Use :func:`mne.pick_info` to pick channels from :class:`mne.Info` (:gh:`10039` by `Mathieu Scheltienne`_)
```